model/src/main/java/io/syndesis/model/connection/Connector.java

```diff
@@ -66,4 +66,9 @@ default Builder builder() {
     class Builder extends ImmutableConnector.Builder {
     }
 
+
+    default Optional<Action> actionById(String id) {
+        return getActions().stream().filter(a -> a.getId().get().equals(id)).findFirst();
```

Comment:
short question (but probably the wrong place here): Why is an actionId an optional when it is mandatory ?

Reply:
Yeah, we have a couple of `Optional`'s in our model that could be revisited, also if it's optional that method should handle that.